### DISCUSSION TITLE
Suggestion adding a new type of cop

### DISCUSSION THREAD

**Discussion by abratashov** [rubocop/rubocop#10473] [Ideas]
Is it possible to add a new class of cop that is able to check the pure business logic that isn't related to codebase text surfing and will be run only one time?
It is needed for adding the extra rule of checking git branch naming convention that is described here https://github.com/rubocop/rubocop/discussions/10470

Example:
```ruby
module Rails
  class GitBranchName < RuboCop::Cop::Cop
    MSG = "Use correct branch name by pattern  '{TicketID}-{Description}'. TicketID is mandatory for linking with the task tracker and should be at least 2 digits"

    def on_send(node = nil)
      branch = `git rev-parse --abbrev-ref HEAD`

      return if starts_from_ticket_number?(branch)

      p "Current branch name: '#{branch}'"

      add_offense(severity: :warning)
    end

    private

    def starts_from_ticket_number?(name)
      gitflow_prefixes = [:bug, :bugfix, :feature, :fix, :hotfix, :origin, :release, :wip]

      name.match?(/(#{gitflow_prefixes.join('/|')})?\d{2,}/)
    end
  end
end
```

**Comment by dvandersluis** [comment]
This isn't really a good usecase for RuboCop IMO. RuboCop is meant for scanning ruby code, and explicitly works on a per-file basis; offenses are tied to files.

I would recommend just creating a script that you call in a git hook or CI, without the overhead of RuboCop.

There also are projects like https://github.com/bkuhlmann/git-lint that might be a better start for what you're trying to do here.

**Reply by abratashov** [reply]
Thank you @dvandersluis, good solution!

**Reply by abratashov** [reply]
I've ended up adding a custom script for checking branch naming and adding it to the CI pipeline `bin/git_check`
https://stackoverflow.com/questions/71601687/how-to-implement-a-custom-rubocop-rule-for-checking-git-branch-naming-convention/71781470#71781470

Also found the hardness of using and maintaining Git hooks due to not all git UI tools  (e.g. SmartGit) correctly supporting hooks (I've gotten unresolved errors, need more time for debugging).
